modelbaker/iliwrapper/ilicache.py

```diff
@@ -115,7 +115,7 @@ def download_repository(self, url):
         Downloads the informationfile (default: ilimodels.xml) and ilisite.xml files from the provided url
         and updates the local cache.
         """
-        netloc = urllib.parse.urlsplit(url)[1] if not os.path.isdir(url) else url
+        netloc = "/".join([urllib.parse.urlsplit(url)[1], urllib.parse.urlsplit(url)[2]]) if not os.path.isdir(url) else url
```

Comment:
This leads to e.g. "geo.so.ch//models/" since `urllib.parse.urlsplit(url)[2]` includes the leading `/` no?

So this might be sufficient: `urllib.parse.urlsplit(url)[1]+urllib.parse.urlsplit(url)[2]`

But I think I would change it anyway, to cover the cases when there are even more levels like "https://geo.so.ch/models/usabilityhub/data/this_is_my_real_repo/models":

I think here something like this makes sense:
```
parsed_url = urllib.parse.urlparse(url)
```
where we receive `geo.so.ch` as `parsed_url.netloc` and `/models` (or `/models/usabilityhub/data/this_is_my_real_repo/models` as `parsed_url.path`.

With this we can create "our" netloc like this:
```
netloc = parsed_url.netloc + parsed_url.path
```

What do you think?

Reply:
Thanks, basically that's fine for me. It works with the geo.so.ch model repository and usability hub as well as with my localhost docker setup.
*But* with the following custom model directories `%ILI_FROM_DB;%XTF_DIR;http://localhost/models/` I get the following warnings in the log:

```
2024-02-09 17:31:35,996 INFO     > ---------- Generate a QGIS Project
2024-02-09 17:31:36,045 INFO    download_repository: netloc: %ILI_FROM_DB
2024-02-09 17:31:36,045 INFO    download_repository: netloc: %XTF_DIR
2024-02-09 17:31:36,046 INFO    download_repository: netloc: localhost/models/
2024-02-09 17:31:36,048 WARNING Could not download %ILI_FROM_DB/ilidata.xml (Das Protokoll "" ist unbekannt)
2024-02-09 17:31:36,048 WARNING Could not download %ILI_FROM_DB/ilisite.xml (Das Protokoll "" ist unbekannt)
2024-02-09 17:31:36,048 WARNING Could not download %XTF_DIR/ilidata.xml (Das Protokoll "" ist unbekannt)
2024-02-09 17:31:36,048 WARNING Could not download %XTF_DIR/ilisite.xml (Das Protokoll "" ist unbekannt)
```

Is this the expected behavior?

Reply:
Yes we don't resolve them in the Model-Selection. We might suppress the request if it's one of those placeholders, or maybe resolve them, but second is not that easy conception-wise (and I think it barely makes sense)

Reply:
Ok, I see.
I've updated ilicache as suggested